### DISCUSSION TITLE
[Noob] How does Whoogle know what to show?

### DISCUSSION THREAD

**Discussion by LiamPerson** [benbusby/whoogle-search#752] [Q&A] [answered]
How does Whoogle know what the Google search results would be and how does it know what to autofill into the search?
Is Whoogle just querying Google directly and if so how does that avoid exposing something like your IP address to Google?

**Comment by Albonycal** [comment, marked as answer]
It's basically like a proxy... whoogle queries google for you.. google will only see the server IP..
and all tracking js etc. is stripped out by whoogle
